### PR TITLE
Issue 87 - Task.loop

### PR DESCRIPTION
**Pull request by frisoft**
Example for `Task.loop`.

### CODE REVIEW

**Comment by Anton-4** [comment]
Thanks for your contribution @frisoft!
I noticed you created this PR as a draft, I assume that means it's not ready for review? Just wanted to make sure so I don't leave you waiting for a review :)

**Comment by frisoft** [comment]
Yes, it's not ready for review, yet, @Anton-4. I want to refactor it a bit and include the proper README.md and tests.
I'll let you know when ready. Thanks

File: examples/TaskLoop/main.roc [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    line <- Stdin.line |> Task.await
+    # Use addNumberFromStdin to calculate the step
+    when addNumberFromStdin total line is
+        # The step has been completed or no further steps to execute
+        Ok stepOrDone -> Task.ok stepOrDone
+        # The inpput cannot be processed. `Task.loop` will return with error
+        Err err -> Task.err err
+
+# This function takes the current state, the total, and a line from stdin and return one of the three possible values
+addNumberFromStdin = \total, line ->
```
Comment (frisoft) [inline examples/TaskLoop/main.roc:32]: I decided to split the logic between `addNumberFromStdinT` and `addNumberFromStdin` functions, with the second to be "Task-free".

This allowed me to test the latter as a pure function. I know `addNumberFromStdinT` is pure as well as it returns just `Task` data, but, to be honest, I don't know how to test it by mocking `Stdin.line` :thinking: .

Is this an acceptable and idiomatic approach or can be done better?

Comment (frisoft) [reply examples/TaskLoop/main.roc:32]: @Anton-4 , this is ready for review, if you have time, please. However, I'm not completely sure about my approach. See prev. comment.

**Comment by frisoft** [comment]
This is ready to review, @Anton-4, when you have time.

**Comment by Anton-4** [comment]
It's in my queue :) I hope to get to it tomorrow

**Comment by Anton-4** [comment]
Today was a busy day, hoping for tomorrow :crossed_fingers:

**Comment by Anton-4** [comment]
I made some language changes in the README, will finish up the review on Friday.
The example is well structured and I think the stdin scenario is a good choice for a simple example!
The Roc code is perhaps a bit heavily commented, for cases were you'd want to use a lot of comments, I generally prefer to introduce the code step by step in the README.md like [here](https://www.roc-lang.org/examples/Tasks/README.html) or [here](https://www.roc-lang.org/examples/RecordBuilder/README.html).

Thanks for your continued contributions @frisoft :)

**Comment by frisoft** [comment]
> I made some language changes in the README, will finish up the review on Friday. The example is well structured and I think the stdin scenario is a good choice for a simple example! The Roc code is perhaps a bit heavily commented, for cases were you'd want to use a lot of comments, I generally prefer to introduce the code step by step in the README.md like [here](https://www.roc-lang.org/examples/Tasks/README.html) or [here](https://www.roc-lang.org/examples/RecordBuilder/README.html).
> 
> Thanks for your continued contributions @frisoft :)

Thank you, @Anton-4, for the changes and the comments. 

I see what you mean by moving the comments to the README.
I'll push a commit to do it.

I'll let you know when the PR is ready for review again.

Thanks

**Comment by frisoft** [comment]
@Anton-4 , I've applied the changes. What do you think?

**Review by Anton-4 — APPROVED** [review]
In my changes I've mainly applied the following principles:
- Shorter and less complex is better: if code or text can be removed or simplified without losing value, we do so.
- Don't make things too beginner friendly; it's not necessary to introduce every part of the code step by step, we should mainly focus on the concept the example is about.

Do not be discouraged by the many changes :)
I've removed and altered some of my own changes as well, through collaborative iteration we make awesome things!

**Comment by frisoft** [comment]
Thank you for the comments, changes and merge, @Anton-4.

**Pull request merged**